code-for-canada/project-thundercat
# [Vulnerability Patch] CWE-471 caused by handlebars@4.0.12

# Description

This patch solves [Prototype Pollution](https://app.snyk.io/vuln/SNYK-JS-HANDLEBARS-173692), related to [CWE-471](https://cwe.mitre.org/data/definitions/471.html) caused by handlebars@4.0.12. Also fixed other dependencies to remove errors and warnings during compilation.

Changes:
- handlebars 4.0.12 -> 4.1.0 (solves High Vulnerability Prototype Pollution)
- react-scripts 2.1.3 -> 2.1.5 (2.1.3 requires earlier version of handlebars)
- added typescript (missing peer dependency for react-scripts)
- removed all devDependencies as unnecessary (versions of eslint conflicted with other deps)

## Type of change

Please delete options that are not relevant.

- Dependency change

## Screenshot

Image of snyk test for frontend after patch
![image](https://user-images.githubusercontent.com/2746350/53654505-de210400-3c1b-11e9-9f30-f3cc2b0272ad.png)


# Testing

Applicable for all code changes.

Manual steps to reproduce this functionality:

1.  Delete frontend/node_modules folder
2. Run docker-compose up
3. Verify that application works as expected
4. Verify that prettier/eslint are still working

Any concerns that need to be addressed in product usability testing:

- I did remove eslint and prettier from devDependencies, but this does not seem to have any effect on the auto-reformatting or any part of our application. Just something to keep in mind, though.

Screenshot of unit tests run passing:
![image](https://user-images.githubusercontent.com/2746350/53654110-e7f63780-3c1a-11e9-814d-5210d668c789.png)


# Checklist

Applicable for all code changes.
